### PR TITLE
refactor(frontend): Refactor imports to use absolute path

### PR DESCRIPTION
**Pull request by yimothysu**
Currently, imports use relative paths. Some components might use `import "../i18n"` while other components will use `import "../../i18n"`. These are actually the same import, but the relative path is different due to different locations in the directory structure.

This PR refactors many imports to use absolute paths for clearer and more consistent imports. For example, no matter where a file is placed in the directory structure, it can `import "src/i18n"` to retrieve translations.

Relative imports are still useful when they actually make use of locality (e.g., tests being in the same directory as a component). Absolute paths are better when the relative location between two files is not meaningful.

### CODE REVIEW

**Comment by amanape** [comment]
Maybe a more suitable convention would be adopting `~` or `#` instead of a real name to prevent (unlikely) conflicts when importing something locally with the same name (e.g., `frontend/src/components/chat/src/interface.tsx`).

Using a symbol might also clarify for developers at first glance which imports are absolute and which are relative, preventing the need for a double-check.

**Comment by amanape** [comment]
Also, consider [vite-tsconfig-paths](https://github.com/aleclarson/vite-tsconfig-paths) for a safer path mapping support with TS/Vite

**Comment by Sparkier** [comment]
> Maybe a more suitable convention would be adopting `~` or `#` instead of a real name to prevent (unlikely) conflicts when importing something locally with the same name (e.g., `frontend/src/components/chat/src/interface.tsx`).
> 
> Using a symbol might also clarify for developers at first glance which imports are absolute and which are relative, preventing the need for a double-check.

Agree but think `~` might not be ideal. Would seem like you're importing from disk root. What [svelte does](https://kit.svelte.dev/docs/modules#$lib) is provide a `$lib` alias, which I like a lot.

**Comment by yimothysu** [comment]
> Also, consider [vite-tsconfig-paths](https://github.com/aleclarson/vite-tsconfig-paths) for a safer and more convenient path mapping support with TS/Vite

I think we already do this. See line 34 in `vite.config.ts`.

**Review by Sparkier — CHANGES_REQUESTED** [review]
Generally good but would implement the two suggested changes from @amanape

**Comment by yimothysu** [comment]
@amanape @Sparkier When I use a symbol like `~/` or `#/`, eslint gives me import errors like the following:

`2:19  error  Missing file extension for "#/utils/toast"  import/extensions`

This must be an eslint issue since the project itself compiles fine. Do you happen to know a solution?

**Comment by amanape** [comment]
> I think we already do this. See line 34 in `vite.config.ts`.

Oh, right. Then the changes you made to `vite.config.ts` may not be required because I believe it is handled by `vite-tsconfig-paths`. You can double-check to confirm

**Comment by amanape** [comment]
@yimothysu It is an [old problem](https://stackoverflow.com/a/59268871/15227904) caused by the `airbnb` config that is setup. It should be safe to add the rule to ignore it as from my knowledge we never (and prefer to not) require adding file extensions to imports

**Comment by yimothysu** [comment]
@amanape works, thanks!

File: frontend/src/components/LoadMessageModal.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import React from "react";
```
Comment (Sparkier) [inline frontend/src/components/LoadMessageModal.tsx:1]: I think this file does not exist anymore.

**Review by Sparkier — CHANGES_REQUESTED** [review]

File: frontend/src/components/LoadMessageModal.tsx (hunk 1 above)
Comment (yimothysu) [reply frontend/src/components/LoadMessageModal.tsx]: Yep, just noticed.

**Pull request merged**